### ISSUE TITLE
Possible race condition in cuckoo map (move/delete)

### ISSUE THREAD

**Issue by EvanKirshenbaum** [HewlettPackard/mpgc#16]
[imported from HPE issue 109]

Refreshing myself on how cuckoo maps work, I think I see an (unlikely) race condition when moving in one thread and deleting in another. Since I had (I thought) walked through all of the possible interleavings, I may be mistaken. Even if not, this may not be worth worrying about before the initial release.

Assume you have a value in the left table and two threads: M, trying to move the entry from left to right, and D trying to delete the entry's key. The interleaving is as follows:

    D: deletes from right (probably not there).
    M: Reads from left.
    M: Copies to right (w/ moving flag)
    M: Checks left (still there)
    D: Deletes from left.
    M: Clears moving flag on right.
    D: Deletes from left (already gone).

Unless I'm missing something, after step 5, the delete has been processed, and the value is gone, but step 6 brings it back, which is incorrect.

[This is an old issue.  It's possible that it's no longer relevant.]

**Comment by EvanKirshenbaum** [comment]
duplicated #6
